### PR TITLE
Add user validation

### LINKED ISSUE #715: Add validation for user creation
### What is the expected behaviour?
Should not be able to create a new user with empty fields.
Also `Create User` button should be disabled until all validations are checked.

### What is the current behaviour?
User can be created with no value passed or only **Name** / **email** is provided.

### How to reproduce the issue?

https://user-images.githubusercontent.com/67645175/132549676-993dde2e-604b-43e6-85d2-d587aafb4a08.mp4

### Screenshots or Screencast
Screenshot 2021-09-08 at 10 07 32 PM

### Any possible solutions?
add validations.

### PR DESCRIPTION
**Pull request by virajbahulkar**
Fixes #715 

Added handleValidation function for validation rules which gets called on form submission.

Loom - 

https://www.loom.com/share/dfe0bdd52471400ab66f4784e37c231d

### CODE REVIEW

**Comment by Navaneeth-pk** [comment]
@gondar00 can you please review this PR?

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx [hunk 1]
```diff
@@ -17,11 +17,77 @@ class ManageOrgUsers extends React.Component {
...
+
+    if (typeof fields['email'] !== 'undefined') {
+      let lastAtPos = fields['email'].lastIndexOf('@');
+      let lastDotPos = fields['email'].lastIndexOf('.');
+
+      if (
+        !(
+          lastAtPos < lastDotPos &&
+          lastAtPos > 0 &&
+          fields['email'].indexOf('@@') == -1 &&
```
Comment (gondar00) [inline frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:72]: I think its `          fields['email'].indexOf('@') == -1 &&`

Comment (virajbahulkar) [reply frontend/src/ManageOrgUsers/ManageOrgUsers.jsx]: @gondar00 I have changed email validation to regular expression now. Please check

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx [hunk 2]
```diff
@@ -17,11 +17,70 @@ class ManageOrgUsers extends React.Component {
...
+        errors['email'] = 'Email is not valid';
+    }
+
+    if (!fields['role']) {
+      formIsValid = false;
+      errors['role'] = 'This field is required';
+    }
+
+    this.setState({ errors: errors });
+    return formIsValid;
```
Comment (gondar00) [inline frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:81]:
```suggestion

    //Name
    if (!fields['firstName']) {
      errors['firstName'] = 'This field is required';
    } else if (typeof fields['firstName'] !== 'undefined') {
      if (!fields['firstName'].match(/^[a-zA-Z]+$/)) {
        errors['firstName'] = 'Only letters are allowed';
      }
    }

    if (!fields['lastName']) {
      errors['lastName'] = 'This field is required';
    } else if (typeof fields['lastName'] !== 'undefined') {
      if (!fields['lastName'].match(/^[a-zA-Z]+$/)) {
        errors['lastName'] = 'Only letters are allowed';
      }
    }

    //Email
    if (!fields['email']) {
      errors['email'] = 'This field is required';
    } else if (!this.validateEmail(fields['email'])) {
        errors['email'] = 'Email is not valid';
    }

    if (!fields['role']) {
      errors['role'] = 'This field is required';
    }

    this.setState({ errors: errors });
    return Object.keys(errors).length  === 0;
```

**Review by gondar00 — CHANGES_REQUESTED** [review]

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx [hunk 3]
```diff
@@ -17,11 +17,57 @@ class ManageOrgUsers extends React.Component {
...
+  validateEmail(email) {
+    console.log(email);
+    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(String(email).toLowerCase());
+  } 
+
+  handleValidation() {
+    let fields = this.state.fields;
+    let errors = {};
+    let formIsValid = true;
```
Comment (gondar00) [inline frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:37]: This can be removed 👍

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx [hunk 4]
```diff
@@ -17,11 +17,57 @@ class ManageOrgUsers extends React.Component {
...
+    }
+    if (!fields['role']) {
+      errors['role'] = 'This field is required';
+    }
+    
+    this.setState({ errors: errors });
+    return Object.keys(errors).length  === 0;
+
+    this.setState({ errors: errors });
+    return formIsValid;
```
Comment (gondar00) [inline frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:68]: also these can be removed 👍

File: frontend/src/ManageOrgUsers/ManageOrgUsers.jsx [hunk 5]
```diff
@@ -79,24 +124,44 @@ class ManageOrgUsers extends React.Component {
...
+  createUser = (event) => {
+    event.preventDefault();
 
-    const { firstName, lastName, email, role } = this.state.newUser;
+    if (this.handleValidation()) {
+      let fields = {};
+      fields['firstName'] = '';
+      fields['lastName'] = '';
+      fields['email'] = '';
+      fields['role'] = '';
```
Comment (gondar00) [inline frontend/src/ManageOrgUsers/ManageOrgUsers.jsx:135]:
```suggestion
      Object.keys(fields).map(key => { fields[key] = '' })
```

**Review by gondar00 — CHANGES_REQUESTED** [review]

**Review by gondar00 — APPROVED** [review]

**Comment by virajbahulkar** [comment]
@Navaneeth-pk please check the heroku link https://user-validation-tooljet.herokuapp.com/

**Pull request merged**